Breeding-Insight/bi-web
# [BI-1401] - Fixing showAll for pagination beyond page 2

# Description
**Story:** [“Show All” for programs pagination doesn’t work if user is on the last page of pagination
](https://breedinginsight.atlassian.net/browse/BI-1401)

Logic for toggling showAll failed on pages beyond page 2 (and only really worked on page 2 due to sheer luck of math). This change passes the correct page and pageSize information to the backend when showAll is toggled, regardless of starting page.

In addition, logic was added to ensure that when a new entry was added when the showAll toggle was selected, the pageSize increases by 1 so it does not spill over onto a second page.

This affects the following tables:

- AdminProgramsTable.vue
- AdminUsersTable.vue
- OntologyTable.vue
- ProgramLocationsTable.vue
- ProgramUsersTable.vue
- TraitsImportTable.vue

GermplasmTable.vue and GermplasmListTable.vue also converted to use BackendPaginationController instead of PaginationController for the sake of consistency and maintainability.

# Dependencies
bi-api/develop

# Testing
For aforementioned affected tables (at minimum one from the first list and then GermplasmTable.vue):

- Change page size such that there are greater than 2 pages
- Navigate to a page greater than 2
- Click showAll (toggling on)
- All values should be displayed and pagination should be on page 1
- Click showAll (toggling off)
- Previous page size should be restored with current page being page 1
- Click showAll (toggling on)
- Change page size
- List should display chosen page size on page 1 with showAll toggled off

# Checklist:

- [X] I have run TAF: [_\<link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/2496577400)
